pkg/externalresource/manager/service.go

```diff
@@ -54,5 +167,199 @@ func (s *Service) CreateResource(
 		return nil, st.Err()
 	}
 
+	// Note that the cache is updated AFTER the metastore is updated.
+	if _, exists := s.cache[request.GetResourceId()]; exists {
+		log.L().Panic("Cache is inconsistent",
```

Comment:
we use upsert for metastore, so even if we panic here, the ground truth has been changed and after another resource manager takes over the service will continue. It's OK that we only warn here?

Reply:
This line seems to be longer needed.